src/position/position.js

```diff
@@ -7,6 +7,8 @@ angular.module('ui.bootstrap.position', [])
  * typeahead suggestions etc.).
  */
   .factory('$position', ['$document', '$window', function ($document, $window) {
+    var POSITION_REGEX = /((top|bottom)(\s+(left|right))?)|(left|right)/,
```

Comment:
There are non-capturing groups, which could be useful in cleaning up your logic below. For example: `/(?:(top|bottom)(?:\s+(left|right))?)|(left|right)/` means that `match[1]` will be top/bottom and `match[2]` will be right/left (only if there's top/bottom), and `match[3]` will be right/left without top/bottom.